File: 8_kyu/Thinkful_Logic_Drills_Traffic_light.jl
"""
  Returns the next traffic light color.
"""
function updatelight(current::String)::String
  i = findfirst(it -> it == current, TRAFFIC_LIGHT_COLORS)
  # Return "red" as error fallback
  TRAFFIC_LIGHT_COLORS[i == nothing ? 3 : (i%length(TRAFFIC_LIGHT_COLORS)+1)]
end

# List of all traffic light colors.
# Order in the tuple is the time sequence of the signales.
const TRAFFIC_LIGHT_COLORS = ("green","yellow","red")

File: 8_kyu/Thinkful_Logic_Drills_Traffic_light.r
# Returns next traffic light color name.
update_light <- function(current) traffic_light_next[current][[1]]

# Definition of the traffic light sequence.
#   name is the current traffic light.
#   value is the next traffic light color.
traffic_light_next = c(green='yellow', yellow='red', red='green')
